src/test/kotlin/com/openlattice/testing/KotlinShuttleTest.kt

```diff
@@ -0,0 +1,98 @@
+package com.openlattice.testing
+
+import com.google.common.base.Stopwatch
+import com.google.common.collect.Queues
+import com.google.common.io.Resources
+import com.openlattice.data.integration.AddressedDataHolder
+import com.openlattice.data.integration.StorageDestination
+import com.openlattice.shuttle.MissionControl
+import com.openlattice.shuttle.payload.SimplePayload
+import com.openlattice.shuttle.test.ShuttleTest.getFlight
+import org.junit.Assert
+import org.junit.Test
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicLong
+import java.util.stream.Stream
+import kotlin.streams.asSequence
+
+class KotlinShuttleTest {
+
+    private val uploadingExecutor = Executors.newSingleThreadExecutor()
+    private val url = Resources.getResource( "cyphers.csv" )
+    private var payload = SimplePayload( url.path )
+
+    @Test( timeout = 5000 )
+    fun testFoo () {
+
+        val flight = getFlight();
+
+        Assert.assertEquals(1, 1)
+
+        val integratedEntities = mutableMapOf<StorageDestination, AtomicLong>().withDefault { AtomicLong(0L) }
+        val integratedEdges = mutableMapOf<StorageDestination, AtomicLong>().withDefault { AtomicLong(0L) }
+        val integrationQueue = Queues
+                .newArrayBlockingQueue<List<Map<String, Any>>>(
+                        Math.max(2, 2 * (Runtime.getRuntime().availableProcessors() - 2))
+                )
+        val rows = AtomicLong(0)
+        val sw = Stopwatch.createStarted()
+        val remaining = AtomicLong(0)
+
+        uploadingExecutor.execute {
+            Stream.generate { integrationQueue.take() }.parallel()
+                    .map { batch ->
+                        println("mapping batch")
+                        try {
+                            rows.addAndGet(batch.size.toLong())
+                            return@map impulse()
+                        } catch (ex: Exception) {
+                            println("caught splosion")
+                            remaining.decrementAndGet()
+                            return@map AddressedDataHolder(mutableMapOf(), mutableMapOf())
```

Comment:
We should fail integration ***OR*** based on the exception that gets propagated engage in a retry strategy. For now lets just fail the integration with really good logging.